libusb/libusb.h

```diff
@@ -953,6 +957,56 @@ struct libusb_ss_usb_device_capability_descriptor {
 	uint16_t bU2DevExitLat;
 };
 
+/** \ingroup libusb_desc
+ * A enum to extract attributes from
+ * \ref libusb_ssplus_usb_device_capability_descriptor.bmAttributes
+ */
+enum libusb_ssplus_attribute_mask {
+	LIBUSB_SSPLUS_ATTRIBUTE_SSCA_MASK = 0x0F,
+	LIBUSB_SSPLUS_ATTRIBUTE_SSIC_MASK = 0xF0
+};
+
+/** \ingroup libusb_desc
+ * A structure representing the SuperSpeedPlus descriptor
+ * This descriptor is documented in section 9.6.2.5 of the USB 3.1 specification.
+ */
+struct libusb_ssplus_usb_device_capability_descriptor {
+	/** Size of this descriptor (in bytes) */
+	uint8_t  bLength;
+	
+	/** Descriptor type. Will have value
+	 * \ref libusb_descriptor_type::LIBUSB_DT_DEVICE_CAPABILITY
+	 * LIBUSB_DT_DEVICE_CAPABILITY in this context. */
+	uint8_t  bDescriptorType;
+	
+	/** Capability type. Will have value
+	 * \ref libusb_capability_type::LIBUSB_BT_SUPERSPEED_PLUS_CAPABILITY
+	 * LIBUSB_BT_SUPERSPEED_PLUS_CAPABILITY in this context. */
+	uint8_t  bDevCapabilityType;
+	
+	/** Reserved field */
+	uint8_t  bReserved;
+	
+	/** Bitmap encoding of:
+	 * ssac: (num sublink speed attributes)
+	 * ssic: (num sublink speed id)
+	 * See enum libusb_ssplus_attribute_mask for convenient masks*/
+	uint32_t bmAttributes;
+	
+	/** Bitmap encoding of supported device level functionalities.
+	 *  ssid, min RX lanes, min TX lanes.
+	 */
+	uint16_t wFunctionalitySupport;
+	
+	/** Reserved field */
+	uint16_t wReserved;
+	
+	/** An array of sublink speed attributes. See \ref bmAttributes for
+	 * the number of attributes  (ssac+1).
+	 */
+	uint32_t* bmSublinkSpeedAttr;
```

Comment:
Had another thought here: could this be a flexible array member, since it's at the end and variable length? Because then we could tag its length a la https://github.com/libusb/libusb/pull/1409

Reply:
I have no opinion about this. I will let the admin decice.

Reply:
Yes this is gone. I asked what the maintainer preferred and they opted for a struct exposing parsed values (vs raw values from the descriptor).